graphql_compiler/compiler/ir_lowering_match/utils.py

```diff
@@ -44,7 +50,7 @@ def convert_coerce_type_and_add_to_where_block(coerce_type_block, where_block):
         return instanceof_filter
 
 
-def expression_list_to_conjunction(expression_list):
+def expression_list_to_conjunction(expression_list: List[BinaryComposition]) -> Expression:
     """Convert a list of expressions to an Expression that is the conjunction of all of them."""
```

Comment:
Good catch, this function is essentially duplicated in two adjacent modules, so I'll delete the one in `between_lowering.py` and use this one from `utils.py` instead.

I'll need to change `List` to `Collection` because I'll need a covariant container (`List` is [invariant](https://mypy.readthedocs.io/en/stable/generics.html#variance-of-generic-types) so `List[BinaryComposition]` is not considered a subtype of `List[Expression]`).

I'll also need to rewrite some tests because the two functions produce a different order of clauses (left-first vs right-first).